jwplatform/client.py

```diff
@@ -219,28 +256,88 @@ def clip(self, site_id, channel_id, event_id, body=None, query_params=None):
         )
 
 
-class _MediaClient(_SiteResourceClient):
+CREATE_MEDIA_PAYLOAD = {
+    "upload": {
+    },
+    "metadata": {
+
+    }
+}
+
 
+class _MediaClient(_SiteResourceClient):
     _resource_name = "media"
     _id_name = "media_id"
 
+    def __init__(self, client: JWPlatformClient):
+        super().__init__(client)
+
     def reupload(self, site_id, body, query_params=None, **kwargs):
         resource_id = kwargs[self._id_name]
         return self._client.request(
             method="PUT",
-            path=self._singular_path.format(site_id=site_id, resource_name=self._resource_name, resource_id=resource_id) + "reupload/",
+            path=self._singular_path.format(site_id=site_id, resource_name=self._resource_name,
+                                            resource_id=resource_id) + "reupload/",
             body=body,
             query_params=query_params
         )
 
-class _WebhookClient(_ResourceClient):
+    def determine_upload_method(self, file, target_part_size) -> str:
+        file_size = common.get_file_size(file)
+        if file_size <= target_part_size:
+            return UploadType.direct.value
+        return UploadType.multipart.value
+
+    def get_upload_handler(self, site_id, file, body=None, query_params=None, **kwargs):
+        if not kwargs:
+            kwargs = {}
 
+        # Determine the upload type - Single or multi-part
+        target_part_size = int(kwargs['target_part_size']) if 'target_part_size' in kwargs else constants.MIN_PART_SIZE
+        upload_method = self.determine_upload_method(file, target_part_size)
+        if not body:
+            body = CREATE_MEDIA_PAYLOAD.copy()
+
+        if 'upload' not in body:
+            body['upload'] = {}
+
+        if not isinstance(body['upload'], dict):
+            raise ValueError(f"Invalid payload structure. The upload element needs to be dictionary.")
+
+        body["upload"]["method"] = upload_method
+
+        # Create the media
+        resp = self.create(site_id, body, query_params)
```

Comment:
What about resuming uploads for already created media?

Reply:
I'm not sure it makes sense to necessarily do anything for media creation here when there is already a method to do so. `determine_upload_method` also doesn't communicate that this actually creates media. It might be useful to step back and define what the user actions we want to support should look like or write an example.

Reply:
So I think it makes sense for this to be a high-level "create media and upload file" method, since that will be the typical flow. Resuming uploads will need to use the lower level upload primitives. I do agree that `get_upload_handler` isn't very clear, I think calling it something like `upload` or `create_and_upload` should communicate that a media is created.